exonum/src/storage/indexes_metadata.rs

```diff
@@ -196,6 +215,51 @@ mod tests {
         let _: ProofMapIndex<_, [u8; 32], i32> = ProofMapIndex::new("test_index", &mut fork);
     }
 
+    #[test]
+    #[should_panic(expected = "Attempt to access index family 'test_index' \
+    while it's an ordinary index")]
+    fn invalid_index_type_in_family() {
+        let database = MemoryDB::new();
+        let mut fork = database.fork();
+        let index_id: i32 = 42;
+        {
+            let mut index = MapIndex::new("test_index", &mut fork);
+            index.put(&"KEY".to_owned(), 42);
+        }
+
+        let _: MapIndex<_, String, i32> =
+            MapIndex::new_in_family("test_index", &index_id, &mut fork);
+    }
+
+    #[test]
+    #[should_panic(expected = "Attempt to access an ordinary index 'test_index' \
+    while it's index family")]
+    fn invalid_index_type_in_family2() {
```

Comment:
This test should be renamed. :upside_down_face: 

It can be named `index_in_family_as_ordinary_index` and the previous test can be named `ordinary_index_as_index_in_family`.